datasets/Fejza_2023/compose.py

```diff
@@ -0,0 +1,161 @@
+import utils
+
+# Read input
+wos1 = pd.read_excel(
+    "https://zenodo.org/records/8348665/files/ECM&ICIs&marker_wos.xls?download=1"
+)
+wos2 = pd.read_excel(
+    "https://zenodo.org/records/8348665/files/ECM&ICIs_wos.xls?download=1"
+)
+wos3 = pd.read_excel(
+    "https://zenodo.org/records/8348665/files/matrixRemodeling&ICIS_wos.xls?download=1"
+)
+wos4 = pd.read_excel(
+    "https://zenodo.org/records/8348665/files/matrixRemodeling&immunoth_wos.xls?download=1"
+)
+wos = pd.concat([wos1, wos2, wos3, wos4])
+wos = utils.extract_doi(wos, "DOI")
+wos = utils.extract_pmid(wos, "Pubmed Id")
+wos = utils.rename_columns(wos, title="Article Title", year="Publication Year")
+
+sco1 = pd.read_csv(
+    "https://zenodo.org/records/8348665/files/ECM&ICIs_scopus.csv?download=1"
+)
+sco3 = pd.read_csv(
+    "https://zenodo.org/records/8348665/files/ECM&immunother_scopus.csv?download=1"
+)
+sco4 = pd.read_csv(
+    "https://zenodo.org/records/8348665/files/matrixRemodeling&ICIs_scopus.csv?download=1"
+)
+sco5 = pd.read_csv(
+    "https://zenodo.org/records/8348665/files/matrixRemodeling&immunother_scopus.csv?download=1"
+)
+
+sco = pd.concat([sco1, sco3, sco4, sco5])
+sco = utils.extract_doi(sco, "DOI")
+sco = utils.rename_columns(sco, title="Title", year="Year")
+
+sco2 = pd.read_csv(
+    "https://zenodo.org/records/8348665/files/ECM&immunoth&marker_scopus.csv?download=1",
+    sep="\t",
+    encoding = "ISO-8859-1"
+)
+sco2 = utils.extract_doi(sco2, "DOI", "", "&")
+sco2 = utils.extract_doi(sco2, "DOI.1", "", "&")
+sco2 = utils.extract_pmid(sco2, "PMID")
+for i in range(25):
+    if i not in [8, 11, 12]:
+        column_name = "Unnamed: " + str(i + 15)
+        sco2 = utils.extract_doi(sco2, column_name, "", "&")
+
+pubs = [
+    "https://zenodo.org/records/8348665/files/summary-ECM&ICI_pubmed.txt?download=1",
+    "https://zenodo.org/records/8348665/files/summary-ECM&immunother_pubmed.txt?download=1",
+    "https://zenodo.org/records/8348665/files/summary-ECMremodeling&ICI_pubmed.txt?download=1",
+    "https://zenodo.org/records/8348665/files/summary-ICI&ECM&marker_pubmed.txt?download=1",
+    "https://zenodo.org/records/8348665/files/summary-matrix_remodeling&immunother_pubmed.txt?download=1",
+]
+
+pmids = []
+
+# files are in pubmed txt file, we aim to just extract the pmid
+for filename in pubs:
+    file = urlopen(filename)
+    current_line = ""
+    current_nr = 1
+    # loop over lines and concatenate separate entries to 1 line so we can extract from it
+    for line in file:
+        line = str(line)
+        line = line.replace("b'", "")
+        line = line.replace('b"', "")
+        line = line.replace("\\r\\n'", " ")
+        if line.startswith(str(current_nr) + ": "):
+            if current_nr > 1:
+                # To the extraction for previous line
+                if "PMID" in current_line:
+                    pm = re.search("PMID: (\d+)", current_line)
+                if pm:
+                    pmids.append(pm.group(1))
```

Comment:
`if pm:` will raise an error if `"PMID"` is not in `current_line`, because `pm` will never have been assigned